modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/impl/MetaStorageManagerImpl.java

```diff
@@ -400,7 +402,12 @@ private CompletableFuture<TopologyAwareRaftGroupService> startRaftNode(
                 RaftGroupEventsListener.noopLsnr,
                 disruptorConfig,
                 topologyAwareRaftGroupServiceFactory,
-                raftGroupOptionsConfigurer
+                options -> {
+                    raftGroupOptionsConfigurer.configure(options);
+
+                    RaftGroupOptions groupOptions = (RaftGroupOptions) options;
+                    groupOptions.externallyEstablishedPeers(metaStorageNodes);
```

Comment:
Both the original solution with flag and the one with externallyEstablishedPeers doesn't fully solve the ABA issue:
1. Original MG.peers = [A] MG.learners = [B,C]
2. [A] dies.
3. Reset MG peers to [B]
4. [B] becomes new leader.
5. [B] dies.
6. Reset MG peers to [C]
7. [C] becomes new leader.
5. [B] returns.
6. As learner [B] accepts the log from [C].
7. [C] dies.
8. Reset MG peers to [B] //At this moment within [B].log there are [... CfgEntry(B::from step 3), ..., CfgEntry(C::from step 6), ..., CfgEntry(B::from step 8), ...]
9. [A] starts with externallyEstablishedPeers([D]) however it will consider CfgEntry(B::from step 3) as matching topology.

Nothing bad will happen in given example and I'm not sure whether bad things are even possible on top of ABA in case of MG, however more robust solution is preferable. Something based on waiting truly externallyEstablishedPeers. Just thoughts though.

Reply:
The approach has been changed, now we are waiting for an index, which solves the ABA problem